scripts/saveState.js

```diff
@@ -65,7 +65,11 @@ function addToSandwich(type, item){
             
         }
 		sidebarBody().find("#sideSubgoal" + item.id).unbind( "click" ).click(function(){
-			drawSubgoal(item.id);
+			// do not enter drawSubgoal with a different id until the current subgoal is saved
+			var isSetSubgoalQuestions = (statusIsTrue("gotSubgoalQuestions"));
+			if (isSetSubgoalQuestions || item.id == subArr.lengh){
```

Comment:
Just for my understanding, why do we need this second check (item.id == subArr.length)?